File: app/src/main/java/com/worldventures/dreamtrips/modules/tripsimages/model/VideoMediaEntity.java
package com.worldventures.dreamtrips.modules.tripsimages.model;

import android.os.Parcel;

import com.worldventures.dreamtrips.modules.feed.model.video.Video;

public class VideoMediaEntity extends BaseMediaEntity<Video> {

   @Override
   public int describeContents() { return 0; }

   @Override
   public void writeToParcel(Parcel dest, int flags) {
      dest.writeInt(this.type == null ? -1 : this.type.ordinal());
      dest.writeParcelable(this.item, 0);
   }

   public VideoMediaEntity() {
      type = TripImageType.VIDEO;
   }

   protected VideoMediaEntity(Parcel in) {
      int tmpType = in.readInt();
      this.type = tmpType == -1 ? null : TripImageType.values()[tmpType];
      this.item = in.readParcelable(Video.class.getClassLoader());
   }

   public static final Creator<VideoMediaEntity> CREATOR = new Creator<VideoMediaEntity>() {
      @Override
      public VideoMediaEntity createFromParcel(Parcel source) {return new VideoMediaEntity(source);}

      @Override
      public VideoMediaEntity[] newArray(int size) {return new VideoMediaEntity[size];}
   };
}

File: app/src/main/java/com/worldventures/dreamtrips/social/ui/tripsimages/model/VideoMediaEntity.kt
package com.worldventures.dreamtrips.social.ui.tripsimages.model

import android.os.Parcel
import android.os.Parcelable
import com.worldventures.dreamtrips.social.ui.feed.model.video.Video

class VideoMediaEntity(val video: Video) : BaseMediaEntity<Video>(video, MediaEntityType.VIDEO), Parcelable {

   constructor(parcel: Parcel) : this(parcel.readParcelable<Video>(Video::class.java.classLoader))

   override fun describeContents(): Int = 0

   override fun writeToParcel(dest: Parcel, flags: Int) = dest.writeParcelable(video, flags)

   companion object CREATOR : Parcelable.Creator<VideoMediaEntity> {
      override fun createFromParcel(parcel: Parcel) = VideoMediaEntity(parcel)
      override fun newArray(size: Int): Array<VideoMediaEntity?> = arrayOfNulls(size)
   }
}
